src/ChangeStream/ResourceStack.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\ChangeStream;
+
+use Puli\Repository\Api\Resource\PuliResource;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use RuntimeException;
+use Webmozart\Assert\Assert;
+
+/**
+ * Represents a versionned stack of resources for a given path.
+ *
+ * You can access different versions of a resource using a ChangeStream
+ * that will return you resource stacks.
+ *
+ * @since  1.0
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class ResourceStack extends ArrayResourceCollection
```

Comment:
I guess you're right.

In this case, though, do we want others to be able to manipulate the stack? (`set()`, `remove()`, ...) Or should we rather not extend anything, store the resources as array and simply provide a `toArray()` method which can be used to convert the stack to an array or a collection?